### PR TITLE
feat: automated AGRS activation

### PR DESCRIPTION
**Pull request by brotherlymite**

### CODE REVIEW

File: diffs/AaveV3EthereumLido_AutomatedAGRSActivation_20241108_before_AaveV3EthereumLido_AutomatedAGRSActivation_20241108_after.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+## Emodes changed
```
Comment (eboadom) [inline diffs/AaveV3EthereumLido_AutomatedAGRSActivation_20241108_before_AaveV3EthereumLido_AutomatedAGRSActivation_20241108_after.md:1]: We need to get this solved, not showing diffs if there are none

File: src/20241108_AaveV3EthereumLido_AutomatedAGRSActivation/AutomatedAGRSActivation.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+As suggested by Chaos Labs, the new automated AGRS system will be configured with the following params:
+
+| Parameter                 | Percent change allowed | minimumDelay |
+| ------------------------- | ---------------------- | ------------ |
+| Base Variable Borrow Rate | 0.5% (absolute change) | 1 day        |
+| Slope 1                   | 0.5% (absolute change) | 1 day        |
+| Slope 2                   | 5% (absolute change)   | 1 day        |
+| Optimal Point (Kink)      | 3% (absolute change)   | 1 day        |
+
+## References
```
Comment (eboadom) [inline src/20241108_AaveV3EthereumLido_AutomatedAGRSActivation/AutomatedAGRSActivation.md:45]: Let's add more explicit links to:
- The injector
- The Edge oracle
- The EdgeRiskSteward

File: src/20241108_AaveV3EthereumLido_AutomatedAGRSActivation/AutomatedAGRSActivation.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+- Recommendations of these parameters will be submitted to [RiskOracle](https://github.com/ChaosLabsInc/risk-oracle/blob/main/src/RiskOracle.sol) smart contract, from the Edge off-chain infrastructure.
+- Between the risk oracle smart contract and the AGRS contract, there will be a very thin middleware [AaveStewardsInjector](https://etherscan.io/address/0x834a5aC6e9D05b92F599A031941262F761c34859), which will have the following logic:
+  - Will take recommendations from the Edge Risk Oracle side and propagate them to the AGRS contract.
+  - Enforce that only the WETH asset can be acted upon.
+  - Given the protections (percentage constraints and time delay) on the AGRS side and that it is an assumption that risk recommendation will be the time correct on the Edge Risk Oracle, the propagation will be permissionless.
+
+The [AaveStewardsInjector](https://etherscan.io/address/0x834a5aC6e9D05b92F599A031941262F761c34859) middle-ware contract will run on Chainlink Automation, and similar to other Aave Robots will be registered using the [AaveCLRobotOperator](https://etherscan.io/address/0x1cDF8879eC8bE012bA959EB515b11008E0cb6323) contract with 1500 LINK from the Ethereum Collector.
```
Comment (eboadom) [inline src/20241108_AaveV3EthereumLido_AutomatedAGRSActivation/AutomatedAGRSActivation.md:32]:
```suggestion
The [AaveStewardsInjector](https://etherscan.io/address/0x834a5aC6e9D05b92F599A031941262F761c34859) middleware, technically being part of the Aave Robot infrastructure, will run on Chainlink Automation and will be registered using the [AaveCLRobotOperator](https://etherscan.io/address/0x1cDF8879eC8bE012bA959EB515b11008E0cb6323) contract with 1500 LINK from the Ethereum Collector.
```

File: src/20241108_AaveV3EthereumLido_AutomatedAGRSActivation/AaveV3EthereumLido_AutomatedAGRSActivation_20241108.sol [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+ * - Discussion: https://governance.aave.com/t/arfc-aave-generalized-risk-stewards-agrs-activation/19178/3
+ * - Snapshot: https://snapshot.org/#/aave.eth/proposal/0x4809f179e517e5745ec13eba8f40d98dab73ca65f8a141bd2f18cc16dcd0cc16
+ */
+contract AaveV3EthereumLido_AutomatedAGRSActivation_20241108 is IProposalGenericExecutor {
+  using SafeERC20 for IERC20;
+  using SafeCast for uint256;
+
+  address public constant EDGE_RISK_STEWARD = 0x81aFd0F99c2Afa2f2DD7E387c2Ef9CD2a29b6E1A;
+  address public constant AAVE_STEWARD_INJECTOR = 0x834a5aC6e9D05b92F599A031941262F761c34859;
+  uint256 public constant LINK_AMOUNT = 1500 ether;
```
Comment (eboadom) [inline src/20241108_AaveV3EthereumLido_AutomatedAGRSActivation/AaveV3EthereumLido_AutomatedAGRSActivation_20241108.sol:25]: Seems too much, no? How much is the cost of each update in gas?

File: src/20241108_AaveV3EthereumLido_AutomatedAGRSActivation/AaveV3EthereumLido_AutomatedAGRSActivation_20241108.sol [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+  function execute() external {
+    AaveV3EthereumLido.ACL_MANAGER.addRiskAdmin(EDGE_RISK_STEWARD);
+
+    AaveV3EthereumLido.COLLECTOR.transfer(
```
Comment (eboadom) [inline src/20241108_AaveV3EthereumLido_AutomatedAGRSActivation/AaveV3EthereumLido_AutomatedAGRSActivation_20241108.sol:30]: Didn't we have helpers for this on the collector utils?

File: src/20241108_AaveV3EthereumLido_AutomatedAGRSActivation/AutomatedAGRSActivation.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
+---
+title: "Automated AGRS Activation"
```
Comment (eboadom) [inline src/20241108_AaveV3EthereumLido_AutomatedAGRSActivation/AutomatedAGRSActivation.md:2]:
```suggestion
title: "Automated (Edge) AGRS Activation"
```

File: src/20241108_AaveV3EthereumLido_AutomatedAGRSActivation/AutomatedAGRSActivation.md [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+author: "BGD Labs (@bgdlabs)"
+discussions: "https://governance.aave.com/t/arfc-aave-generalized-risk-stewards-agrs-activation/19178/3"
+snapshot: "https://snapshot.org/#/aave.eth/proposal/0x4809f179e517e5745ec13eba8f40d98dab73ca65f8a141bd2f18cc16dcd0cc16"
+---
+
+## Simple Summary
+
+This proposal activates the automated Aave Generalized Risk Stewards (AGRS) system on the Aave Ethereum Lido Instance to perform automated interest rate updates for the WETH asset.
```
Comment (eboadom) [inline src/20241108_AaveV3EthereumLido_AutomatedAGRSActivation/AutomatedAGRSActivation.md:10]:
```suggestion
This proposal activates the automated Aave Generalized Risk Stewards (AGRS) system on the Aave Ethereum Lido Instance to perform automated interest rate updates for the WETH asset.
Under the hood, the AGRS consumes the Edge infrastructure by Chaos Labs.
```

File: src/20241108_AaveV3EthereumLido_AutomatedAGRSActivation/AaveV3EthereumLido_AutomatedAGRSActivation_20241108.sol (hunk 5 above)
Comment (brotherlymite) [reply src/20241108_AaveV3EthereumLido_AutomatedAGRSActivation/AaveV3EthereumLido_AutomatedAGRSActivation_20241108.sol]: Fixed

File: src/20241108_AaveV3EthereumLido_AutomatedAGRSActivation/AaveV3EthereumLido_AutomatedAGRSActivation_20241108.sol (hunk 4 above)
Comment (brotherlymite) [reply src/20241108_AaveV3EthereumLido_AutomatedAGRSActivation/AaveV3EthereumLido_AutomatedAGRSActivation_20241108.sol]: Makes sense, have reduced the amount

**Review by eboadom — APPROVED** [review]

**Pull request merged**